core/src/main/java/org/apache/carbondata/core/datastore/page/LazyColumnPage.java

```diff
@@ -93,7 +93,22 @@ public double getDouble(int rowId) {
 
   @Override
   public float getFloat(int rowId) {
-    throw new UnsupportedOperationException("internal error");
+    DataType dataType = columnPage.getDataType();
```

Comment:
Just call `getDouble(rowId)` from this method and then typecast it. No need to duplicate the code

Reply:
okay